src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -1974,4 +2058,413 @@ pub mod tests {
             must_commit(&mut engine, key, 60, 69);
         }
     }
+
+    #[test]
+    fn test_lock_with_conflict_should_not_exist() {
+        let mut engine = TestEngineBuilder::new().build().unwrap();
+
+        must_prewrite_put(&mut engine, b"k1", b"v1", b"k1", 20);
+        must_commit(&mut engine, b"k1", 20, 30);
+
+        // Key already exists.
+        let e = acquire_pessimistic_lock_allow_lock_with_conflict(
+            &mut engine,
+            b"k1",
+            b"k1",
+            10,
+            10,
+            false,
+            false,
+            true,
+            false,
+        )
+        .unwrap_err();
+        match e {
+            MvccError(box ErrorInner::WriteConflict { .. }) => (),
+            e => panic!("unexpected error: {:?}", e),
+        }
+        must_unlocked(&mut engine, b"k1");
+
+        // Key already exists and already locked by the same txn.
```

Comment:
The impact is small, make it clear first is ok.

Reply:
Previously it will return locked_with_conflict in conflicting case, which doesn't seem to cause any correctness issue. Here we unified its behavior. For not conflicting case, it's what [this issue](https://github.com/pingcap/tidb/issues/40114) discussed about.